libs/platform/src/lib/components/carosuel/carosuel.component.ts

```diff
@@ -0,0 +1,526 @@
+        const direction = Direction.NEXT;
+        const indexEl = this.findNextSlideIndex(direction, force);
+        if (videoList.length > 0) {
+            // Check for video carousel
+            for (let i = 0; i < videoList.length; i++) {
+                if (i === indexEl) {
+                    videoList[i].play();
+                } else {
+                    videoList[i].pause();
+                }
+            }
+        }
+        if (this.animation === 'slide') {
+            this.pause();
+            this.slideAnimation(this.findNextSlideIndex(Direction.NEXT, force), Direction.NEXT);
+            this.cdRef.markForCheck();
+        } else if (this.animation === 'fade') {
+            this.pause();
+            this.fadeAnimation(this.findNextSlideIndex(Direction.NEXT, force), Direction.NEXT);
+            this.cdRef.markForCheck();
+        } else {
+            this.activeSlide = this.findNextSlideIndex(Direction.NEXT, force);
+            this.cdRef.markForCheck();
+        }
+        if (!this.animation) {
+            this.activeSlideChange.emit({ direction: 'Next', relatedTarget: this.activeSlide });
+        }
+
+    }
+    /** To display previous slide*/
+    public previousSlide(force: boolean = false): void {
+        this.restartTimer();
+        // Start previous slide, pause actual slide
+        const videoList = this.el.nativeElement.getElementsByTagName('video');
+        const direction = Direction.PREV;
+        const indexel = this.findNextSlideIndex(direction, force);
+        if (videoList.length > 0) {
+            // Check for video carousel
+            for (let i = 0; i < videoList.length; i++) {
+                if (i === indexel) {
+                    videoList[i].play();
+                } else {
+                    videoList[i].pause();
+                }
+            }
+        }
+
+
+        if (this.animation === 'slide') {
+            this.pause();
+            this.slideAnimation(this.findNextSlideIndex(direction, force), direction);
+            this.cdRef.markForCheck();
+        } else if (this.animation === 'fade') {
+            this.pause();
+            this.fadeAnimation(this.findNextSlideIndex(Direction.PREV, force), Direction.PREV);
+            this.cdRef.markForCheck();
+        } else {
+            this.activeSlide = this.findNextSlideIndex(Direction.PREV, force);
+            this.cdRef.markForCheck();
+        }
+        if (!this.animation) {
+            this.activeSlideChange.emit({ direction: 'Prev', relatedTarget: this.activeSlide });
+        }
+    }
+
+    /**Display slides in fade animation mode */
+    protected fadeAnimation(goToIndex: number, direction?: any) {
+        const goToSlide = this.slides[goToIndex];
+
+        if (this.animationEnd) {
+            this.animationEnd = false;
+
+            goToSlide.directionNext = true;
+            if (this.isBrowser) {
+                setTimeout(() => {
```

Comment:
You should store references to all `setTimeout` calls, and call `clearTimeout` on all references in `ngOnDestroy` for component. This will prevent potential memory leaks.